pkg/services/control/server/evacuate.go

```diff
@@ -39,3 +49,62 @@ func (s *Server) EvacuateShard(_ context.Context, req *control.EvacuateShardRequ
 	}
 	return resp, nil
 }
+
+func (s *Server) replicate(addr oid.Address, obj *objectSDK.Object) error {
+	nm, err := s.netMapSrc.GetNetMap(0)
+	if err != nil {
+		return err
+	}
+
+	cid, ok := obj.ContainerID()
+	if !ok {
+		// Container ID is expected to be here. If for some reason it is not,
+		// we log error and return nil to prevent situations
+		// where a shard can't be evacuated because of a single bad/corrupted object.
+		return nil
+	}
+
+	c, err := s.cnrSrc.Get(cid)
+	if err != nil {
+		return err
+	}
+
+	binCnr := make([]byte, sha256.Size)
+	cid.Encode(binCnr)
+
+	ns, err := nm.ContainerNodes(c.Value.PlacementPolicy(), binCnr)
+	if err != nil {
+		return fmt.Errorf("can't build a list of container nodes")
+	}
+
+	nodes := placement.FlattenNodes(ns)
+	bs := (*keys.PublicKey)(&s.key.PublicKey).Bytes()
+	for i := 0; i < len(nodes); i++ {
+		if bytes.Equal(nodes[i].PublicKey(), bs) {
+			copy(nodes[i:], nodes[i+1:])
+			nodes = nodes[:len(nodes)-1]
+		}
+	}
+
+	var res replicatorResult
```

Comment:
what if `nodes` is empty after the current node exclusion?

Reply:
The `res.count == 0` after `HandleTask` and we will return an error.

Reply:
`HandleTask` has some debug logging even with an empty task so i meant it could be cached earlier, but ok